app/controllers/application_controller.rb

```diff
@@ -1,17 +1,27 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
 
-  before_action :find_user
+  before_action :find_user, :require_login
 
   def render_404
     # DPR: this will actually render a 404 page in production
     raise ActionController::RoutingError.new('Not Found')
   end
 
-private
+  private
   def find_user
     if session[:user_id]
       @login_user = User.find_by(id: session[:user_id])
     end
   end
+
+
+  def require_login
+    # if @login_user.nil?
+    #   flash[:error] = "You must be logged in to view this"
+    #   redirect_to root_path
```

Comment:
I love the idea of putting this in a controller filter - why is this commented out?